src/index/index.c

```diff
@@ -11,80 +11,147 @@
 #include "../query_ctx.h"
 #include "../util/rmalloc.h"
 #include "../datatypes/point.h"
-#include "../graph/graphcontext.h"
-#include "../graph/entities/node.h"
-#include "../graph/rg_matrix/rg_matrix_iter.h"
+#include "../datatypes/vector.h"
 
 #include <stdatomic.h>
 
+// returns a type aware name for a given field
+// e.g.
+// "range:age" for a range field named "age"
+// "vector:location" for a vector field named "location"
+// "name" for a full text field named "name"
+void Index_TypedFieldName
+(
+	char *type_aware_name,  // [out] type aware name
+	IndexFieldType t,       // field type
+	const char *name        // field name
+) {
+	ASSERT(name != NULL);
+	ASSERT(type_aware_name != NULL);
+	ASSERT(t == INDEX_FLD_FULLTEXT ||
+		   t == INDEX_FLD_RANGE    ||
+		   t == INDEX_FLD_VECTOR);
+
+	if(t == INDEX_FLD_FULLTEXT) {
```

Comment:
move to switch case

Reply:
I don't believe I can as `INDEX_FLD_RANGE` isn't an enum value but a macro oring 3 different enum values.
`switch case` can't have `INDEX_FLD_RANGE` as one of its cases